scenes/quests/lore_quests/quest_001/2_ink_combat/components/fill_game_logic.gd

```diff
@@ -20,6 +20,11 @@ func start() -> void:
 	_update_allowed_colors()
 
 
+func _ready() -> void:
+	var filling_barrels: Array = get_tree().get_nodes_in_group("filling_barrels")
+	barrels_to_win = clampi(barrels_to_win, 0, filling_barrels.size())
```

Comment:
Do we want to emit `goal_reached` immediately (or perhaps deferred) in the case where `barrels_to_win` is `0`? Or log a warning?